client/src/utils/ajax.ts

```diff
@@ -51,16 +55,93 @@ async function request<T>(
 /** GET **/
 
 interface SessionUser {
-  user: UserType;
+  user: { [username: string]: UserType };
   sessionMeta: { activeDonations: number };
-  result: string;
 }
-export function getSessionUser(): Promise<SessionUser> {
-  return get('/user/get-session-user');
+
+type challengeFilesForFiles = {
+  files: Array<Omit<ChallengeFile, 'fileKey'> & { key: string }>;
+} & Omit<CompletedChallenge, 'challengeFiles'>;
+
+type ApiSessionResponse = Omit<SessionUser, 'user'>;
+type ApiUser = {
+  user: {
+    [username: string]: ApiUserType;
+  };
+  result: string;
+};
+
+type ApiUserType = Omit<UserType, 'completedChallenges'> & {
+  completedChallenges: challengeFilesForFiles[];
+};
+
+type UserResponseType = {
+  user: { [username: string]: UserType };
+  result: string;
+};
+
+function parseApiResponseToClientUser<T>(
+  data: ApiUser | (T & ApiUser)
+): UserResponseType {
```

Comment:
```suggestion
function parseApiResponseToClientUser(data: ApiUser): UserResponseType {
```
T doesn't seem to be used.

Reply:
T is used here: `L113: const response: Promise<ApiUser & ApiSessionResponse>` -> This is passed as `data`.

Reply:
`ApiUser & ApiSessionResponse` is passed, sure, but `T` isn't used.  There are no calls to `parseApiResponseToClientUser<AType>`.

What's happening here is just that `ApiUser & ApiSessionResponse` can be used everywhere `ApiUser` can, `parseApiResponseToClientUser` just ignores the `ApiSessionResponse` properties.